test/testBrowserLanguage.js

```diff
@@ -0,0 +1,112 @@
+import browserLanguage from '../src/nginxconfig/util/browser_language';
+
+class mockLocales{
+    static languages = [];
+
+    static language = null;
+    IntlBackup = null;
+
+    setNavigatorLanguages(langs){
+        mockLocales.languages = langs;
+        return this;
+    }
+
+    setNavigatorLanguage(lang){
+        mockLocales.language = lang;
+        return this;
+    }
+
+    setDateTimeLocale(locale) {
+        const newDateTimeForMat = new Intl.DateTimeFormat(locale);
+        this.IntlBackup = Intl;
+        if (!locale){
+            // eslint-disable-next-line no-global-assign
+            Intl = undefined;
+            return;
+        }
+
+        // eslint-disable-next-line no-global-assign
+        Intl = {
+            DateTimeFormat(){
+                return newDateTimeForMat;
+            },
+        };
+        return this;
+    }
+
+    restoreDateTimeLocale(){
+        // eslint-disable-next-line no-global-assign
+        Intl = this.IntlBackup;
+        return this;
+    }
```

Comment:
There is an odd mix of static and non-static here -- considering this touches globals, I see no reason to need to instantiate anything here, we should just make everything static?